clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -7628,6 +7628,8 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
   if (getToolChain().getTriple().isSYCLDeviceEnvironment()) {
     TranslatorArgs.push_back("-spirv-max-version=1.1");
     std::string ExtArg("-spirv-ext=+all");
+    if (C.getArgs().hasArg(options::OPT_fsycl_esimd))
+      TranslatorArgs.push_back("-spirv-allow-unknown-intrinsics");
```

Comment:
Could you please add a quick test be to `clang/test/Driver/sycl-offload.c`? Something like:
```
/// RUN: %clangxx %s -fsycl -fsycl-esimd -### 2>&1 | FileCheck %s --check-prefix=CHK-FSYCL-ESIMD
/// CHK-FSYCL-ESIMD: llvm-spirv {{.*}} "-spirv-allow-unknown-intrinsics"
```
Sorry for missing out on this ealier.